crates/sui-indexer-alt-graphql/src/pagination.rs

```diff
@@ -290,9 +290,77 @@ pub(crate) fn is_connection(field: &MetaField) -> bool {
     type_.ends_with("Connection") || type_.ends_with("Connection!")
 }
 
+/// Ensures ASC order DESC order, then applies pagination limits with overhead for paginate_results to check
+/// for has_prev and has_next page. Returns items in ascending order as GraphQL cursor syntax expects items
+/// to be in ascending order.
+///
+/// ```
+/// // Example usage:
+/// let events: Vec<Event> = /* ... */;
+/// let results = ordered_items_with_limits(
+///     &page,
+///     events,
+///     |e| (e.tx_sequence_number, e.sequence_number)
+/// );
+/// ```
+pub(crate) fn ordered_results_with_limits<Result, SortKey, KeyFn, Cursor>(
+    page: &Page<Cursor>,
+    results: impl IntoIterator<Item = Result>,
+    key_fn: KeyFn,
+) -> Vec<Result>
+where
+    SortKey: Ord,
+    KeyFn: Fn(&Result) -> SortKey,
+{
+    let mut results: Vec<Result> = results.into_iter().collect();
```

Comment:
If the first thing you do is turn the results into a vector, you might as well accept a `Vec<Result>` directly.

Reply:
Noted, going to remove this function and have the results ordering handled by the function creating the Events.